### PR TITLE
[Cache] remove deprecated PSR-16 implementations et al.

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

\o/

(use `Psr16Cache` instead)

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/ArrayAdapter.php [hunk 1]
```diff
@@ -156,8 +193,110 @@ public function commit()
...
+
+        return $values;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset()
+    {
+        $this->clear();
```
Comment (Simperfit) [inline src/Symfony/Component/Cache/Adapter/ArrayAdapter.php]: isn't just doing the same thing as clear without the return better ?

Comment (nicolas-grekas) [reply src/Symfony/Component/Cache/Adapter/ArrayAdapter.php]: We must implement both: clear() for psr-6, reset() for service-contracts.
This is copy/paste from existing current implementations that exist in traits.

**Review by Simperfit — APPROVED** [review]

File: src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php [hunk 2]
```diff
@@ -32,6 +47,292 @@ class MemcachedAdapter extends AbstractAdapter
...
+     * Examples for servers:
+     * - 'memcached://user:pass@localhost?weight=33'
+     * - [['localhost', 11211, 33]]
+     *
+     * @param array[]|string|string[] $servers An array of servers, a DSN, or an array of DSNs
+     * @param array                   $options An array of options
+     *
+     * @return \Memcached
+     *
+     * @throws \ErrorException When invalid options or servers are provided
```
Comment (samnela) [inline src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php]: @throws  \ErrorException or InvalidArgumentException ?

Comment (nicolas-grekas) [reply src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php]: If you found a mismatch, please submit a fix on branch 4.2. this PR is only moving code around.

**Review by Tobion — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php (hunk 2 above)
Comment (samnela) [reply src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php]: Ok, thanks for the explanation.

**Pull request merged**